sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -71,6 +72,25 @@ def test_f_classif():
     assert_array_almost_equal(pv_sparse, pv)
 
 
+@pytest.mark.parametrize("coeff", [abs_r_regression, r_regression])
+def test_r_regression(coeff):
+    X, y = make_regression(n_samples=200, n_features=20, n_informative=5,
+                           shuffle=False, random_state=0)
+
+    pearson_r = coeff(X, y)
+    assert ((pearson_r < 1).all())
+
+    # with centering, compare with sparse
+    pearson_r = coeff(X, y, center=True)
+    pearson_r_sparse = coeff(sparse.csr_matrix(X), y, center=True)
+    assert_array_almost_equal(pearson_r_sparse, pearson_r)
+
+    # again without centering, compare with sparse
+    pearson_r = coeff(X, y, center=False)
+    pearson_r_sparse = coeff(sparse.csr_matrix(X), y, center=False)
+    assert_array_almost_equal(pearson_r_sparse, pearson_r)
+
+
```

Comment:
Please test the equivalence with the ordering returned by `f_regression`'s F statistics, preferably on a a wide dataset (e.g. with n_samples=200 and n_features=1000) where feature selection is typically used:

```python
assert_array_equal(f_regression(X, y)[0].argsort(),
                   abs_r_regression(X, y).argsort())
```

and please also check:

Please add a test to check that:

```python
assert_allclose(abs_r_regression(X, y), np.abs(r_regression(X, y)))
```

possibly in the same test, e.g. named `test_f_regression_r_regression_consistency`.